app/src/main/java/com/asfoundation/wallet/ui/onboarding/OnboardingActivity.kt

```diff
@@ -68,8 +80,14 @@ class OnboardingActivity : BaseActivity(), OnboardingView {
     linkSubject = PublishSubject.create()
     presenter = OnboardingPresenter(CompositeDisposable(), this, interactor,
         AndroidSchedulers.mainThread(), smsValidationInteract, Schedulers.io(),
-        ReplaySubject.create(), referralInteractor)
-    setupUI()
+        ReplaySubject.create(), referralInteractor, repository)
+    if (savedInstanceState != null && savedInstanceState.containsKey(PAYMENT_METHODS_ICONS)) {
+      setupUI(savedInstanceState.getStringArrayList(PAYMENT_METHODS_ICONS)!!
+          .toList())
+    } else {
+      setupUI()
+    }
```

Comment:
i think this logic shouldn't be in oncreate. move it to setupUI or create a new private method for that